src/tokens/selectors.ts

```diff
@@ -356,7 +343,26 @@ export const swappableTokensByNetworkIdSelector = createSelector(
           (tokenInfo.minimumAppVersionToSwap &&
             !isVersionBelowMinimum(appVersion, tokenInfo.minimumAppVersionToSwap))
       )
-      .sort(tokenCompareByUsdBalanceThenByName)
+      .sort((token1, token2) => {
+        // treat prices without priceUsd separately
+        if (token1.priceUsd === null || token2.priceUsd === null) {
+          // If both prices are null, sort alphabetically by name
+          if (!token1.priceUsd && !token2.priceUsd) {
+            return token1.name.localeCompare(token2.name)
+          }
+          // Otherwise, sort such that the token with a non-null price comes first
+          return token1.priceUsd === null ? 1 : -1
+        }
+
+        // Sort by balance (higher balances first)
+        const token1UsdBalance = token1.balance.multipliedBy(token1.priceUsd)
+        const token2UsdBalance = token2.balance.multipliedBy(token2.priceUsd)
+        if (token1UsdBalance > token2UsdBalance) return -1
```

Comment:
I think this converts the values to `number` loosing precision and should be avoided.

Reply:
Here's a problematic case:

<img width="835" alt="Screenshot 2024-01-11 at 13 52 15" src="https://github.com/valora-inc/wallet/assets/57791/d34df780-6faf-4367-8853-d8a54d67903c">

Reply:
yikes, i was treating this as a number for some reason. updated to use the big number comparisons, thanks for the catch!